### PR TITLE
This fixes issue #34. Using the wrong hostname.

### PR DESCRIPTION
**Pull request by gdey**
When registering as a slave we are using the wrong hostname. It should
be our hostname, not the hostname name of the server we are connecting
to.

### CODE REVIEW

File: replication/binlogsyncer.go [hunk 1]
```diff
@@ -62,6 +65,15 @@ func NewBinlogSyncer(serverID uint32, flavor string) *BinlogSyncer {
 	return b
 }
 
+func (b *BinlogSyncer) LocalHostname() string {
+
+	if b.MyHost == "" {
```
Comment (siddontang) [inline replication/binlogsyncer.go]: where is MyHost?

File: replication/binlogsyncer.go [hunk 2]
```diff
@@ -25,10 +26,12 @@ type BinlogSyncer struct {
 	c        *client.Conn
 	serverID uint32
 
-	host     string
-	port     uint16
-	user     string
-	password string
+	// LocalHost is the name of you want to present to the MySQL master. If it is not set it will default to os.Hostname()
+	LocalHost string
```
Comment (siddontang) [inline replication/binlogsyncer.go]: I see we don't use LocalHost anywhere.

**Comment by siddontang** [comment]
Hi @gdey , I guess if we don't register slave, we can still sync binlog from mysql master.
Maybe we can remove register slave later?

File: replication/binlogsyncer.go (hunk 1 above)
Comment (gdey) [inline replication/binlogsyncer.go]: Sorry, that should have been LocalHost, not MyHost. I started with MyHost, then change my mind on the name. I just missed this when I redid the changes on my local computer. I'm surprised that the compiler did not catch this?

**Comment by gdey** [comment]
I'm going to close this and reopen it. Goofed in my repo. And started committing to master, instead of a feature branch. I'm not sure what you are asking with the second comment.

**Pull request closed without merging**